File: python/Graph/Bellman_Ford.py
def bellmanFord(V, edges, src):
  dist = [float('inf')] * V 
  dist[src] = 0 

  for _ in range(V - 1):
    for u, v, wt in edges:
      if dist[u] != float('inf') and dist[u] + wt < dist[v]:
        dist[v] = dist[u] + wt 
  
  return dist

File: C++/Graphs/Bellman_Ford.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> bellman_ford(int V, vector<vector<int>> &edges, int S)
{
  vector<int> dist(V, 1e8);
  dist[S] = 0;

  for (int i = 0; i < V - 1; i++)
  {
    for (auto edge : edges)
    {
      int u = edge[0];
      int v = edge[1];
      int wt = edge[2];

      if (dist[u] != 1e8 && dist[u] + wt < dist[v])
        dist[v] = dist[u] + wt;
    }
  }
  return dist;
}